### PR TITLE
Change Config to Typescript

### PR DESCRIPTION
**Pull request by ericyliu**
Description
-----------
- adds typing for config
- changes `config.example.js` to typescript
- adds runtime typechecking for config through [runtypes](https://github.com/pelotom/runtypes)

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// Server configuration
```
Comment (jkeat) [inline config.ts:1]: should this be tracked?

Comment (ericyliu) [reply config.ts:1]: .. no good catch, my bad

File: tsconfig.json [hunk 2]
```diff
@@ -2,5 +2,6 @@
   "compilerOptions": {
       "esModuleInterop": true,
       "allowUmdGlobalAccess": true,
-  }
+      "outDir": "build",
+  },
```
Comment (jkeat) [inline tsconfig.json:6]: @ericyliu quick typescript question -- when do builds come into play? It looks like a regular run (`npm run start`) doesn't involve building to this dir

Comment (ericyliu) [reply tsconfig.json:6]: Yeah that's a good question. I realized when running `tsc --watch` it'll compile the files and generate js files at the same location as the ts ones. This is so all the generated js files go into a specific folder. It doesn't come into play when running it in prod

Comment (ericyliu) [reply tsconfig.json:6]: Yeah that's a good question. I realized when running `tsc --watch` it'll compile the files and generate js files at the same location as the ts ones. This is so all the generated js files go into a specific folder. It doesn't come into play when running it in prod

**Pull request merged**